### DISCUSSION TITLE
How do I get 'credential' of google's response?

### DISCUSSION THREAD

**Discussion by jaber-al-nahian-klikit** [cuongdevjs/reactjs-social-login#129] [Q&A] [answered]
With then following code:

``
 function handleCallbackResponse(response) {
    console.log('rep', response);
    console.log(response.credential)
  }

  useEffect(() => {
    /* global google */
    google.accounts.id.initialize({
      client_id: "223764948932-jq8fb6pcv1ku78srsmrtkbjq5haq9pcc.apps.googleusercontent.com",
      callback: handleCallbackResponse
    })

    google.accounts.id.renderButton(
      document.getElementById("signInBtn"),
      {
        theme: "outline", size: "large"
      }
    )

  }, []) 
  ``

  
  I can get credential string like below.

Example:
credential: "eyJhbGciOiJSUzI1NiIsImtpZCI6ImM3ZTExNDEwNTlhMTliMjE4MjA5YmM1YWY3YTgxYTcyMGUzOWI1MDAiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJhenAiOiIyMjM3NjQ5NDg5MzItanE4ZmI2cGN2MWt1NzhzcnNtcnRrYmpxNWhhcTlwY2MuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhdWQiOiIyMjM3NjQ5NDg5MzItanE4ZmI2cGN2MWt1NzhzcnNtcnRrYmpxNWhhcTlwY2MuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJzdWIiOiIxMDgwNTAwMTgxNzQ0NjI5ODM0ODIiLCJoZCI6InZpdmFjb21zb2x1dGlvbnMuY29tIiwiZW1haWwiOiJqYWJlci5uYWhpYW5Adml2YWNvbXNvbHV0aW9ucy5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwibmJmIjoxNjkzODI3MDI1LCJuYW1lIjoiSmFiZXIgQWwgTmFoaWFuIiwicGljdHVyZSI6Imh0dHBzOi8vbGgzLmdvb2dsZXVzZXJjb250ZW50LmNvbS9hL0FBY0hUdGRVWldwWWdwWjBveWpKTkEwNG1iMUpVTElELTVuTW1wc3lDOXZTT2M4M1VqUT1zOTYtYyIsImdpdmVuX25hbWUiOiJKYWJlciIsImZhbWlseV9uYW1lIjoiQWwgTmFoaWFuIiwibG9jYWxlIjoiZW4iLCJpYXQiOjE2OTM4MjczMjUsImV4cCI6MTY5MzgzMDkyNSwianRpIjoiZWQzNWI0ZDljNjJmZmU4ZmZiN2ZhNTE5Yzc4OGM4ZTM2ZjI1MDhhMSJ9.b4-SUBSRUcf6hQ54DBlNXLJRpax7joqMqSDnrwW1QiNqniocAt3ekKohjuAAL2fPDDzDWB7qB6X3dm0Y8fikOopUZOzdCEUmNOHzFtU-NrGFVNxXWOGPUllGCmDWQYXIILv2HLhqG15IvpANrN6yoSfptgpyHu0GRc-nhIMT1IkXmpvALW0UxOhDwUZ84J_rfGb8smQ__Enjbbgd5E2746dA9hQsvBWnNG9Bhb1N0T99FZtvu0Lc9ZQ-C-blJYDVA7vPInByjw"

How do I get this data with your plugin?

**Comment by cuongdevjs** [comment, marked as answer]
There is my sample code:

````
                                        <LoginSocialGoogle
						client_id={}
						onLoginStart={onLoginStart}
						redirect_uri={REDIRECT_URI}
						typeResponse="idToken"
						scope="openid profile email"
						ux_mode="popup"
						onResolve={({ provider, data }) => {
							console.log("🚀 ~ file: App.js:89 ~ App ~ data:", data);
							setProvider(provider);
							setProfile(data);
						}}
						onReject={(err) => {
							console.log(err);
						}}
					>
						<GoogleLoginButton />
					</LoginSocialGoogle>
````
